src/domain/flow-system/src/aggregates/flow_configuration/flow_configuration.rs

```diff
@@ -57,6 +57,34 @@ impl FlowConfiguration {
         self.apply(event)
     }
 
+    /// Pause configuration
+    pub fn pause(
+        &mut self,
+        now: DateTime<Utc>,
+    ) -> Result<(), ProjectionError<FlowConfigurationState>> {
+        let event = FlowConfigurationEventModified {
+            event_time: now,
+            flow_key: self.flow_key.clone(),
+            paused: true,
+            rule: self.rule.clone(),
+        };
+        self.apply(event)
+    }
+
+    /// Resume configuration
+    pub fn resume(
+        &mut self,
+        now: DateTime<Utc>,
+    ) -> Result<(), ProjectionError<FlowConfigurationState>> {
+        let event = FlowConfigurationEventModified {
+            event_time: now,
+            flow_key: self.flow_key.clone(),
+            paused: false,
```

Comment:
Is it worth checking the aggregate state to see if it is already paused/unpaused not to produce redundant events?

Reply:
Agree, makes sense